contrib/epee/include/misc_language.h

```diff
@@ -30,74 +30,14 @@
 
 #include <boost/utility/value_init.hpp>
 #include <boost/shared_ptr.hpp>
-#include <limits>
-#include <functional>
 #include <vector>
 namespace epee
 {
-#define STD_TRY_BEGIN() try {
-
-#define STD_TRY_CATCH(where_, ret_val) \
-	} \
-	catch (const std::exception  &e) \
-	{ \
-		LOG_ERROR("EXCEPTION: " << where_  << ", mes: "<< e.what());  \
-		return ret_val; \
-	} \
-	catch (...) \
-	{ \
-		LOG_ERROR("EXCEPTION: " << where_ ); \
-		return ret_val; \
-	}
-
-
-
 #define AUTO_VAL_INIT(v)   boost::value_initialized<decltype(v)>()
 
 namespace misc_utils
 {
-	template<typename t_type>
-		t_type get_max_t_val(t_type t)
-		{
-			return (std::numeric_limits<t_type>::max)();
-		}
-
-		
-	template<typename t_iterator>
-		t_iterator move_it_forward(t_iterator it, size_t count)
-		{
-			while(count--)
-				it++;
-			return it;
-		}
-
-    template<typename t_iterator>
-    t_iterator move_it_backward(t_iterator it, size_t count)
```

Comment:
I'd leave it.

Reply:
`std::advance` does the same thing, and it also does a compile-time check to see if the iterator is random-access to turn an O(count) operation into a O(1) operation.